src/program/packetblaster/lwaftr/lwaftr.lua

```diff
@@ -0,0 +1,201 @@
+   help         = "h",    -- display help text
+   size         = "S",    -- packet size list (defaults to IMIX)
+   src_mac      = "s",    -- source ethernet address
+   dst_mac      = "d",    -- destination ethernet address
+   b4           = "b",    -- B4 start IPv6_address,IPv4_address,port
+   aftr         = "a",    -- fix AFTR public IPv6_address
+   ipv4         = "I",    -- fix public IPv4 address
+   count        = "c",    -- how many b4 clients to simulate
+   rate         = "r",    -- rate in MPPS (0 => listen only)
+   v4only       = "P",    -- generate only public IPv4 traffic
+   v6only       = "E"     -- generate only public IPv6 encapsulated traffic
+}
+
+local function dir_exists(path)
+  local stat = S.stat(path)
+  return stat and stat.isdir
+end
+
+function run (args)
+   local opt = {}
+   local duration
+   local c = config.new()
+
+   function opt.D (arg)
+      duration = assert(tonumber(arg), "duration is not a number!")
+   end
+
+   local verbose
+   function opt.V (arg)
+      verbose = true
+   end
+
+   function opt.h (arg)
+      print(usage)
+      main.exit(1)
+   end
+
+   local sizes = { 64, 64, 64, 64, 64, 64, 64, 594, 594, 594, 1500 }
+   local sizes_ipv6 = { 104, 104, 104, 104, 104, 104, 104, 634, 634, 634, 1540 }
+   function opt.S (arg)
+     sizes = {}
+     sizes_ipv6 = {}
+     for size in string.gmatch(arg, "%d+") do
+       local s = tonumber(size)
+       if s < 28 then
+         s = 28
+         print("Warning: Increasing IPv4 packet size to 28")
+       end
+       sizes[#sizes+1] = s
+       sizes_ipv6[#sizes_ipv6+1] = s + 40
+     end
+   end
+
+   local src_mac = "00:00:00:00:00:00"
+   function opt.s (arg) src_mac = arg end
+
+   local dst_mac = "00:00:00:00:00:00"
+   function opt.d (arg) dst_mac = arg end
+
+   local b4_ipv6, b4_ipv4, b4_port = "2001:db8::", "10.0.0.0", 1024
+   function opt.b (arg) 
+     for s in string.gmatch(arg, "[%w.:]+") do
+        if string.find(s, ":") then
+          b4_ipv6 = s
+        elseif string.find(s, '.',1,true) then
+          b4_ipv4 = s
+        else
+          b4_port = assert(tonumber(s), string.format("UDP port %s is not a number!", s))
+        end
+     end
+   end
+
+   local public_ipv4 = "8.8.8.8"
+   function opt.I (arg) public_ipv4 = arg end
+
+   local aftr_ipv6 = "2001:db8:ffff::100"
+   function opt.a (arg) aftr_ipv6 = arg end
+
+   local count = 1
+   function opt.c (arg) 
+     count = assert(tonumber(arg), "count is not a number!")
+   end
+
+   local rate = 1
+   function opt.r (arg) 
+     rate = assert(tonumber(arg), "rate is not a number!")
+   end
+
+   local pciaddr
+   function opt.p (arg) 
+     pciaddr = arg
+   end
+
+   local ipv4_only = false
+   function opt.P () ipv4_only = true end
+
+   local ipv6_only = false
+   function opt.E () ipv6_only = true end
+
+   -- TODO how can I use digit options like -4? function opt.4 isn't valid in lua
+   args = lib.dogetopt(args, opt, "VD:hS:s:a:d:b:iI:c:r:PEp:", long_opts)
```

Comment:
In case of using invalid characters such as numbers or a dash, which cannot be used as property names directly, should use the alternative hash syntax using brackets. Example:

```lua
> opts = {} opts["4"] = "hi"
> = opts.4
stdin:1: '<eof>' expected near '.4'
> = opts["4"]
hi
```

That said I don't know if `lib.dogetopt` would be able to parse '4' as a short name option.